src/py21cmfast/drivers/lightcone.py

```diff
@@ -860,32 +843,31 @@ def run_lightcone(
 
     Other Parameters
     ----------------
-    regenerate, write, direc, random_seed, hooks
+    regenerate, write, direc, hooks
         See docs of :func:`initial_conditions` for more information.
     """
     direc, regenerate, hooks = _get_config_options(direc, regenerate, write, hooks)
 
     pf_given = any(perturbed_fields)
     if pf_given and initial_conditions is None:
         raise ValueError(
-            f"If perturbed_fields {perturbed_fields} are provided, initial_conditions {initial_conditions} must be provided"
+            "If perturbed_fields are provided, initial_conditions must be provided"
         )
 
-    # For the high-level, we need all the InputStruct initialised
-    # NOTE: the random seed is now set here instead of in the OutputStruct init
-    #   if it's passed as None
-    # TODO: make sure cosmo_params is consistent with lightconer.cosmo
-    if isinstance(inputs, str):
-        inputs = InputParameters.from_template(inputs, seed=random_seed)
-    elif inputs is None:
-        inputs = InputParameters.from_defaults(seed=random_seed)
-    # For the high-level, we need all the InputStruct initialised
-    cosmo_params = inputs.cosmo_params
-    user_params = inputs.user_params
-    flag_options = inputs.astro_params
-    astro_params = inputs.flag_options
-
-    random_seed = inputs.random_seed
+    # TODO: make sure cosmo_params is consistent with lightconer.cosmo as well
+    inputs = InputParameters.from_output_structs(
+        (initial_conditions,) + perturbed_fields,
+        cosmo_params=inputs.cosmo_params,
+        user_params=inputs.user_params,
+        astro_params=inputs.astro_params,
+        flag_options=inputs.flag_options,
+        random_seed=inputs.random_seed,
+    )
+
+    # if no random seed was given to the inputs, and no output structs were passed,
+    #   raise an error
+    if not isinstance(inputs.random_seed, int):
+        raise ValueError("An integer seed, or initial conditions must be given")
```

Comment:
this would be removed also if we move the random seed validation to the input parameters class.